packages/core/src/App/Containers/Layout/header/traders-hub-home-button.tsx

```diff
@@ -5,21 +5,24 @@ import { Icon, Text } from '@deriv/components';
 import { routes } from '@deriv/shared';
 import { observer, useStore } from '@deriv/stores';
 import { Localize } from '@deriv/translations';
+import { useFeatureFlags } from '@deriv/hooks';
 
 const TradersHubHomeButton = observer(() => {
     const { ui } = useStore();
     const { is_dark_mode_on } = ui;
     const history = useHistory();
     const location = useLocation();
     const { pathname } = location;
+    const { is_next_wallet_enabled } = useFeatureFlags();
 
     return (
         <div
             data-testid='dt_traders_hub_home_button'
             className={classNames('traders-hub-header__tradershub', {
-                'traders-hub-header__tradershub--active': pathname === routes.traders_hub,
+                'traders-hub-header__tradershub--active':
+                    pathname === routes.traders_hub || pathname === routes.wallets,
             })}
-            onClick={() => history.push(routes.traders_hub)}
+            onClick={() => history.push(is_next_wallet_enabled ? routes.wallets : routes.traders_hub)}
```

Comment:
This ensures that when user clicks on `Trader's Hub` in desktop when feature flag is enabled, they are redirected to `/wallets`, otherwise they are directed to `appstore/traders-hub`:

https://github.com/binary-com/deriv-app/assets/103016120/2bbaf0de-0060-49ce-8079-33f8f6a0de83